src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy

```diff
@@ -239,6 +243,52 @@ class PitestPlugin implements Plugin<Project> {
                 dependencies.add(project.dependencies.create(junit5PluginDependencyAsString))
             }
         }
+
+        addJUnitPlatformLauncherDependencyIfNeeded()
+    }
+
+    private void addJUnitPlatformLauncherDependencyIfNeeded() {
+        Configuration testImplementation = project.configurations.findByName("testImplementation")
+        testImplementation.withDependencies { directDependencies ->
+            if (!extension.addJUnitPlatformLauncher.isPresent() || !extension.addJUnitPlatformLauncher.get()) {
```

Comment:
I wonder if `testImplementation.withDependencies` could be executed before called earlier ` pitestConfiguration.withDependencies` adding pitest-junit5-plugin? Or the order of add/call is (and will be) always kept by Gradle?

Reply:
It turns our that it can. The order depends (among others) on the order of Gradle configuration resolving. `testImplementation` seems to be resolved much earlier than `pitest`. However, it should be not a problem, as we add a dependency based on junit-platform libraries, not the PIT plugin for JUnit 5.